File: 724-find-pivot-index/724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        # left = 0
        # right = len(nums)
        # for i in range(len(nums)):
        #     if sum(nums[left:i]) == sum(nums[i+1:right]):
        #         return i
        # return -1
        left, right = 0, sum(nums)
        for index, num in enumerate(nums):
            right -= num
            if left == right:
                return index
            left += num
        return -1
        

File: 724-find-pivot-index/724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let left = 0;
    let right = nums.reduce((total, current) => {return total+current}, 0);
    for (let i=0; i<nums.length; i++){
        right -= nums[i];
        if (right === left){
            return i;
        }
        left += nums[i];
    };
    return -1;
};